strings/base_types.h

```diff
@@ -89,7 +89,7 @@ WINRT_EXPORT namespace winrt
         {
```

Comment:
This function can't be `noexcept` otherwise it will just abort as before.

Also, all of the `hex_to_xxx` functions are noexcept and terminate on failure. They would need to be updated as well.